src/main/java/io/github/programminglife2016/pl1_2016/parser/Data.java

```diff
@@ -0,0 +1,7 @@
+package io.github.programminglife2016.pl1_2016.parser;
+
+/**
+ * Created by ravishivam on 4-5-16.
+ */
+public interface Data {
+}
```

Comment:
An interface without methods is usually not what you want. Ideally every method of the implementations of an interface should override the interface methods. Also, `Data` is not a clear name.